src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceService.kt

```diff
@@ -220,50 +222,113 @@ class LicenceService(
     }
   }
 
+  /**
+   * Add additional condition. Allows for adding more than one condition of the same type
+   */
+  @Transactional
+  fun addAdditionalCondition(
+    licenceId: Long,
+    conditionType: String,
+    request: AddAdditionalConditionRequest
+  ): AdditionalConditionModel {
+    val licenceEntity = licenceRepository
+      .findById(licenceId)
+      .orElseThrow { EntityNotFoundException("$licenceId") }
+
+    val username = SecurityContextHolder.getContext().authentication.name
+
+    val newConditions = licenceEntity.additionalConditions.toMutableList()
+
+    newConditions.add(
+      AdditionalCondition(
+        conditionType = request.conditionType,
+        conditionCode = request.conditionCode,
+        conditionText = request.conditionText,
+        expandedConditionText = request.expandedText,
+        conditionCategory = request.conditionCategory,
+        licence = licenceEntity,
+        conditionSequence = request.sequence
+      )
+    )
+
+    val updatedLicence = licenceEntity.copy(
+      additionalConditions = newConditions,
```

Comment:
I learnt recently that `.plus` or `+` works on immutable lists and returns a new copy:
so I think you could skip intermediate variable:
```
additionalConditions = additionalConditions + newCondition
```